build/phases.go

```diff
@@ -7,25 +7,18 @@ import (
 
 const (
 	layersDir      = "/layers"
-	buildpacksDir  = "/buildpacks"
-	platformDir    = "/platform"
-	orderPath      = "/buildpacks/order.toml"
-	groupPath      = `/layers/group.toml`
-	planPath       = "/layers/plan.toml"
 	appDir         = "/workspace"
 	cacheDir       = "/cache"
 	launchCacheDir = "/launch-cache"
+	platformDir    = "/platform"
 )
 
 func (l *Lifecycle) Detect(ctx context.Context) error {
 	detect, err := l.NewPhase(
 		"detector",
 		WithArgs(
-			"-buildpacks", buildpacksDir,
```

Comment:
If I understand this change correctly, we are no longer explicitly setting paths and strictly relying on the defaults set in the lifecycle for most paths. Is that right? 

If so, why can't the other directories also be defaulted?

Reply:
I left the other directories b/c, for every build, we do things like mount volumes or write env vars to those paths. Therefore, the lifecycle defaults could result in a failure if they don't agree w/ `pack`. `lifecycle` and `buildpacks` paths only matter when you are creating a builder and we have more flexibility if `pack build` doesn't try to override the defaults (unless... you use `--buildpack` during `build` and clean separation breaks down).